Makefile

```diff
@@ -1,59 +1,62 @@
-BSON_PKGS = $(shell etc/list_pkgs.sh ./bson)
-BSON_TEST_PKGS = $(shell etc/list_test_pkgs.sh ./bson)
-EVENT_PKGS = $(shell etc/list_pkgs.sh ./event)
-EVENT_TEST_PKGS = $(shell etc/list_test_pkgs.sh ./event)
-MONGO_PKGS = $(shell etc/list_pkgs.sh ./mongo)
-MONGO_TEST_PKGS = $(shell etc/list_test_pkgs.sh ./mongo)
-UNSTABLE_PKGS = $(shell etc/list_pkgs.sh ./x)
-UNSTABLE_TEST_PKGS = $(shell etc/list_test_pkgs.sh ./x)
-TAG_PKG = $(shell etc/list_pkgs.sh ./tag)
-TAG_TEST_PKG = $(shell etc/list_test_pkgs.sh ./tag)
-EXAMPLES_PKGS = $(shell etc/list_pkgs.sh ./examples)
-EXAMPLES_TEST_PKGS = $(shell etc/list_test_pkgs.sh ./examples)
-PKGS = $(BSON_PKGS) $(EVENT_PKGS) $(MONGO_PKGS) $(UNSTABLE_PKGS) $(TAG_PKG) $(EXAMPLES_PKGS)
-TEST_PKGS = $(BSON_TEST_PKGS) $(EVENT_TEST_PKGS) $(MONGO_TEST_PKGS) $(UNSTABLE_TEST_PKGS) $(TAG_PKG) $(EXAMPLES_TEST_PKGS)
+# We list packages with shell scripts and loop through them to avoid testing with ./...
+# Running go test ./... will run tests in all packages concurrently which can lead to
+# unexpected errors.
+#
+# TODO(GODRIVER-2093): Use ./... to run tests in all packages with parallelism and remove
+# these PKG variables and loops from all make targets.
+PKGS = $(shell etc/list_pkgs.sh)
+TEST_PKGS = $(shell etc/list_test_pkgs.sh)
+
 ATLAS_URIS = "$(ATLAS_FREE)" "$(ATLAS_REPLSET)" "$(ATLAS_SHARD)" "$(ATLAS_TLS11)" "$(ATLAS_TLS12)" "$(ATLAS_FREE_SRV)" "$(ATLAS_REPLSET_SRV)" "$(ATLAS_SHARD_SRV)" "$(ATLAS_TLS11_SRV)" "$(ATLAS_TLS12_SRV)" "$(ATLAS_SERVERLESS)" "$(ATLAS_SERVERLESS_SRV)"
 GODISTS=linux/amd64 linux/386 linux/arm64 linux/arm linux/s390x
-
 TEST_TIMEOUT = 1800
 
+### Utility targets. ###
 .PHONY: default
-default: check-env check-fmt build-examples lint test-cover test-race
+default: add-license build build-examples check-env check-fmt lint test-short
 
-.PHONY: check-env
-check-env:
-	etc/check_env.sh
+.PHONY: add-license
+add-license:
+	# Find all .go files not in the vendor directory and try to write a license notice.
+	find . -path ./vendor -prune -o -type f -name "*.go" -print | xargs ./etc/add_license.sh
+	# Check for any changes made with -G. to ignore permissions changes. Exit with a non-zero
+	# exit code if there is a diff.
+	git diff -G. --quiet
 
-.PHONY: doc
-doc:
-	godoc -http=:6060 -index
+.PHONY: build
+build:
+	go build $(BUILD_TAGS) $(PKGS)
 
 .PHONY: build-examples
 build-examples:
 	go build $(BUILD_TAGS) ./examples/... ./x/mongo/driver/examples/...
 
-.PHONY: build
-build:
-	go build $(BUILD_TAGS) $(filter-out ./core/auth/internal/gssapi,$(PKGS))
-
 .PHONY: build-no-tags
 build-no-tags:
-	go build $(filter-out ./core/auth/internal/gssapi,$(PKGS))
+	go build $(PKGS)
 
 .PHONY: build-tests
 build-tests:
-	for TEST in $(PKGS); do \
+	for TEST in $(TEST_PKGS); do \
 		go test $(BUILD_TAGS) -c $$TEST ; \
 		if [ $$? -ne 0 ]; \
 		then \
 			exit 1; \
 		fi \
 	done
 
+.PHONY: check-env
+check-env:
+	etc/check_env.sh
```

Comment:
Is there still a use case for the `check_env.sh` script? Seems like it would be better for the Go test code that consumes those env variables to validate them, not a separate script.

Reply:
Hmm nothing in our CI seems to use it. I'm happy to remove the script and make target if @qingyang-hu agrees; you don't necessarily need a value for `AUTH`, `SSL` or `MONGO_GO_DRIVER_COMPRESSOR` to run tests, so it's an odd "check".

Reply:
I did not quite get the point. I see these values are used in the test code. Don't we need a sanity check before using them?

Reply:
Thank you for the explanation! It makes sense to remove the script then.